src/app/public/modules/phone-field/phone-field-input.directive.ts

```diff
@@ -0,0 +1,232 @@
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+// tslint:enable
+
+@Directive({
+  selector: '[skyPhoneFieldInput]',
+  providers: [
+    SKY_PHONE_FIELD_VALUE_ACCESSOR,
+    SKY_PHONE_FIELD_VALIDATOR,
+    SkyPhoneFieldAdapterService
+  ]
+})
+export class SkyPhoneFieldInputDirective implements OnInit, AfterViewInit,
+  ControlValueAccessor, Validator {
+
+  /**
+   * Disabling linting here as per the Angular style guide "Style 05-13" it is acceptable
+   * to alias when when the directive name is also an input property,
+   * and the directive name doesn't describe the property.
+   */
+  // tslint:disable-next-line:no-input-rename
+  @Input('skyPhoneFieldInput')
+  public skyPhoneFieldComponent: SkyPhoneFieldComponent;
+
+  @Input()
+  public set defaultCountry(value: string) {
+    this.skyPhoneFieldComponent.setDefaultCountry(value);
+    this._defaultCountryCode = value;
+  }
+
+  public get defaultCountry() {
+    return this._defaultCountryCode;
+  }
+
+  @Input()
+  public set disabled(value: boolean) {
+    this.skyPhoneFieldComponent.disabled = value;
+    this.adapterService.setElementDisabledState(this.elRef.nativeElement, value);
+    this._disabled = value;
+  }
+
+  public get disabled(): boolean {
+    return this._disabled || false;
+  }
+
+  @Input()
+  public formatModel: boolean = true;
+
+  @Input()
+  public noValidate: boolean = false;
+
+  @Output()
+  public selectedCountryChanged = new EventEmitter<SkyCountryData>();
+
+  private set modelValue(value: string) {
+    this._modelValue = value;
+    this.adapterService.setElementValue(this.elRef.nativeElement, value);
+    if (value) {
+      let formattedValue = value;
+      if (this.formatModel) {
+        formattedValue = this.skyPhoneFieldComponent.formatNumber(value.toString());
+      }
+      this._onChange('+' + this.skyPhoneFieldComponent.selectedCountry.dialCode + ' ' + formattedValue);
+    } else {
+      this._onChange(value);
+    }
+    this._validatorChange();
+  }
+
+  private get modelValue(): string {
+    return this._modelValue;
+  }
+
+  private _defaultCountryCode: string;
+
+  private _disabled: boolean;
+
+  private _modelValue: string;
+
+  public constructor(
+    private adapterService: SkyPhoneFieldAdapterService,
+    private changeDetector: ChangeDetectorRef,
+    private elRef: ElementRef,
+    private injector: Injector
+  ) { }
+
+  public ngOnInit(): void {
+    this.adapterService.addElementClass(this.elRef.nativeElement, 'sky-form-control');
+    if (this.defaultCountry) {
+      this.skyPhoneFieldComponent.selectCountry(this.defaultCountry);
+    }
+    this.adapterService.setElementPlaceholder(this.elRef.nativeElement,
+      this.skyPhoneFieldComponent.selectedCountry.placeholder);
+
+    this.adapterService.setAriaLabel(this.elRef.nativeElement);
+  }
+
+  public ngAfterViewInit(): void {
+    this.skyPhoneFieldComponent.selectedCountryChanged.subscribe((country: SkyCountryData) => {
```

Comment:
Should this `.subscribe()` have a `.takeUntil()`?

Reply:
Fixed.